app/Models/Forum/Topic.php

```diff
@@ -195,6 +195,21 @@ public function nthPost($n)
         return $this->posts()->skip(intval($n) - 1)->first();
     }
 
+    public function isDoublePostBy($userId)
+    {
+        $post = $this->posts()->orderBy('post_id', 'desc')->first();
+        $postTime = $post['post_time'];
+        $now = Carbon::now();
+
+        $diffSinceLastPost = $now->diffInDays($postTime);
+
+        if (! is_null($userId) && $post['poster_id'] == $userId && $diffSinceLastPost < 3) {
+            return true;
```

Comment:
Validation in the following order: 
- Check if the passed userId is `null` to rule out errors
- Check if the last poster id is the id that got passed in
- Check if the diff is bigger than 3

Or what is your question?